core/two-steps.ts

```diff
@@ -0,0 +1,45 @@
+import { router } from '@slowreader/core'
+import { atom } from 'nanostores'
+
+interface BackRoutes {
+  [key: string]: string
+}
+
+export type Side = 'first' | 'second'
+
+export const backRoutes: BackRoutes = {
+  categories: '/feeds/categories',
+  fast: '/fast/:category',
+  slow: '/slow/:feed'
+}
+
+export const side = atom<Side>('first')
+
+export function toggleSide(event: KeyboardEvent): void {
+  if (event.key === 'Escape') {
+    switch (side.get()) {
+      case 'first':
+        side.set('second')
+        break
+      case 'second':
+        side.set('first')
+        break
+      default:
+        side.set('first')
+    }
+  }
+}
+
+router.subscribe(route => {
+  if (route.route === 'categories' && route.params.feed) {
+    side.set('second')
+  } else if (route.route === 'fast' && route.params.post) {
+    backRoutes.fast = `/fast/${route.params.category}`
```

Comment:
We can’t use URL here too.

It is better to use `envrionment.openRoute()` which accept just a route object (and environment use URL or any other type of routing).